hikari/events/channel_events.py

```diff
@@ -780,3 +788,264 @@ async def fetch_guild_webhooks(self) -> typing.Sequence[webhooks.PartialWebhook]
+        # <<inherited docstring from GuildThreadEvent>>.
+        return self.thread.id
+
+
+@base_events.requires_intents(intents.Intents.GUILDS)
+@attr_extensions.with_copy
+@attr.define(kw_only=True, weakref_slot=False)
+class GuildThreadUpdateEvent(GuildThreadEvent):
+    """Event fired when a thread is updated."""
+
+    shard: gateway_shard.GatewayShard = attr.field(metadata={attr_extensions.SKIP_DEEP_COPY: True})
+    # <<inherited docstring from ShardEvent>>.
+
+    thread: channels.GuildThreadChannel = attr.field()
+    """The thread that was updated."""
+
+    @property
+    def app(self) -> traits.RESTAware:
+        # <<inherited docstring from Event>>.
+        return self.thread.app
+
+    @property
+    def guild_id(self) -> snowflakes.Snowflake:
+        # <<inherited docstring from GuildThreadEvent>>.
+        return self.thread.guild_id
+
+    @property
+    def thread_id(self) -> snowflakes.Snowflake:
+        # <<inherited docstring from GuildThreadEvent>>.
+        return self.thread.id
+
+
+@base_events.requires_intents(intents.Intents.GUILDS)
+@attr_extensions.with_copy
+@attr.define(kw_only=True, weakref_slot=False)
+class GuildThreadDeleteEvent(GuildThreadEvent):
+    """Event fired when a thread is deleted."""
+
+    app: traits.RESTAware = attr.field(metadata={attr_extensions.SKIP_DEEP_COPY: True})
+    # <<inherited docstring from Event>>.
+
+    shard: gateway_shard.GatewayShard = attr.field(metadata={attr_extensions.SKIP_DEEP_COPY: True})
+    # <<inherited docstring from ShardEvent>>.
+
+    thread_id: snowflakes.Snowflake = attr.field()
+    # <<inherited docstring from GuildThreadEvent>>.
+
+    guild_id: snowflakes.Snowflake = attr.field()
+    # <<inherited docstring from GuildThreadEvent>>.
+
+    parent_id: snowflakes.Snowflake = attr.field()
+    """The ID of the channel that the thread was deleted from."""
+
+    type: channels.ChannelType = attr.field()
+    """The type of thread that was deleted."""
+
+
+@base_events.requires_intents(intents.Intents.GUILDS)
+@attr_extensions.with_copy
+@attr.define(kw_only=True, weakref_slot=False)
+class OwnThreadMemberUpdateEvent(GuildThreadEvent):
+    """Event fired when the bot's own member is updated in a thread."""
+
+    app: traits.RESTAware = attr.field(metadata={attr_extensions.SKIP_DEEP_COPY: True})
+    # <<inherited docstring from Event>>.
+
+    shard: gateway_shard.GatewayShard = attr.field(metadata={attr_extensions.SKIP_DEEP_COPY: True})
+    # <<inherited docstring from ShardEvent>>.
+
+    member: channels.ThreadMember = attr.field()
+    """The member that was updated."""
+
+    guild_id: snowflakes.Snowflake = attr.field()
+    # <<inherited docstring from GuildThreadEvent>>.
+
+    @property
+    def thread_id(self) -> snowflakes.Snowflake:
+        # <<inherited docstring from GuildThreadEvent>>.
+        return self.member.thread_id
+
+
+@base_events.requires_intents(intents.Intents.GUILDS | intents.Intents.GUILD_MEMBERS)
```

Comment:
It is still possible to receive this event even if you dont have the GUILD_MEMBERS intent:
> If the current user does not have the GUILD_MEMBERS [Gateway Intent](https://discord.com/developers/docs/topics/gateway#gateway-intents), then this event will only be sent if the current user was added to or removed from the thread.